hicommonwealth/commonwealth
# [Sign-in Revamp]: New Auth Modal + Updated Wallet behaviours

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6384

## Description of Changes
- Adds new Auth Modal UI wth responsive screen behaviours
- Integrated Cosmos/Substrate and Solana wallets in the new Modal i.e Polkadot, Phantom, Leap, and Keplr. EVM wallets will be integrated in https://github.com/hicommonwealth/commonwealth/issues/6387
- Integrated SSO options in the new modal i.e Google, X (Twitter), Discord, and Github. Email login will be integrated in https://github.com/hicommonwealth/commonwealth/issues/6386
- Updated behaviours for wallet options visibility.
  - EVM wallet options (+ WalletConnect) will now only show in EVM communities, in all non-community pages and as requested by the `showWalletsFor` in the new modal
  - Cosmos wallet options will now only show in Cosmos communities, in all non-community pages and as requested by the `showWalletsFor` in the new modal.
  - Substrate wallet options will now only show in Substrate communities, in all non-community pages and as requested by the `showWalletsFor` in the new modal
  - Solana wallet options will now only show in Solana communities, in all non-community pages and as requested by the `showWalletsFor` in the new modal
- Now during a new wallet account signup, we no longer require the user to input their username/profile-pic, instead we directly create an account for them with the default `Anonymous` name
- Updated all instances of the old `LoginModal` with the new `AuthModal`
- New `AuthModal` is behind the new `FLAG_NEW_SIGN_IN_MODAL` flag, to see changes enable the flag in `.env`

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_NEW_SIGN_IN_MODAL` in `.env`
- Install and setup `Keplr`, `Leap`, `Phantom` and `Polkadot` wallets if you don't have already.
- Logout (if logged in) and go to any non-community page, open the sign-in modal and verify you see all available wallets for all 4 supported chains (EVM, Cosmos, Solana, Substrate) in the wallets tab -> `Wallet Connect`, `Keplr` and `Leap`, `Phantom` and `Polkadot`, verify that signing in with all wallets (excluding wallet connect) works in those pages. Also verify that signing in with all SSO's works in those pages
- Logout (if logged in) and go to any Solana based community (ex: `/solana/discussions`), open the sign-in modal and verify you see only the `Phantom` wallet in the wallets tab, verify that signing in `Phantom` wallet and all `SSO's` works in those pages.
- Logout (if logged in) and go to any Substrate based community (ex: `/edgeware/discussions`), open the sign-in modal and verify you see only the `Polkadot` wallet  in the wallets tab, verify that signing in `Polkadot` wallet and all `SSO's` works in those pages.
- Logout (if logged in) and go to any Ethereum based community (ex: `/dydx/discussions`), open the sign-in modal and verify you see only the `WalletConnect` option  in the wallets tab, verify that signing in with all `SSO's` works in those pages. (WalletConnect will be integrated later)
- Logout (if logged in) and go to new community page (`/createCommunity`), click on any option i.e `Ethereum`, `Cosmos` or `Polygon` and verify that `Ethereum` + `Polygon` when clicked opens the `AuthModal` with only the `WalletConnect` option in the wallets tab, and `Cosmos` option when clicked opens the `AuthModal` with only the `Leap` and `Keplr` options in the wallets tab. All SSO's will be available in the SSO's tab. Verify that signing-in with different wallets works in this page.

## Deployment Plan
Part of https://github.com/hicommonwealth/commonwealth/milestone/38

## Other Considerations
`FLAG_NEW_SIGN_IN_MODAL` should not be enabled in prod yet.
